dataverse-webapp/src/main/webapp/editSingleFile.xhtml

```diff
@@ -0,0 +1,381 @@
+                                         update=":fileForm:fileTable"
+                        />
+                    </div>
+                </p:dialog>
+                <!--Set dataset thumbnail from file window-->
+                <p:dialog id="fileSetThumbnail" styleClass="smallPopUp" header="#{bundle['file.setThumbnail.header']}" widgetVar="fileSetThumbnail" modal="true">
+                    <div class="form-horizontal">
+                        <div class="form-group">
+                            <label for="datasetThumbnailImage" class="col-sm-4 control-label">
+                                #{bundle['file.datasetThumbnail']}
+                            </label>
+                            <div class="col-sm-8">
+                                <p class="help-block">#{bundle['file.datasetThumbnail.tip']}</p>
+                                <p:selectBooleanCheckbox id="datasetThumbnailImage"
+                                                         itemLabel="#{bundle['file.useThisIamge']}" value="#{EditSingleFilePage.useAsDatasetThumbnail}"/>
+                                <p:message for="datasetThumbnailImage" display="text" />
+                            </div>
+                        </div>
+                    </div>
+                    <div class="button-block">
+                        <p:commandButton id="fileSetThumbnailSaveBtn"
+                                         value="#{bundle.saveChanges}"
+                                         oncomplete="PF('fileSetThumbnail').hide()"
+                                         update=":fileForm:fileTable,:messagePanel"
+                                         actionListener="#{EditSingleFilePage.saveAsDesignatedThumbnail()}"/>
+                        <p:commandButton id="fileSetThumbnailCancelBtn"
+                                         value="#{bundle.cancel}"
+                                         onclick="PF('fileSetThumbnail').hide();PF('blockFileForm').hide();"
+                        />
+                    </div>
+                </p:dialog>
+                <!--Confirm thumbnail change on modal window-->
+                <p:dialog id="fileThumbnailConfirm" styleClass="smallPopUp" header="#{bundle['file.setThumbnail.header']}" widgetVar="fileThumbnailConfirm" modal="true">
+                    <div>
+                        <p class="help-block">
+                            <span class="glyphicon glyphicon-exclamation-sign text-danger"/> <span class="text-danger">#{bundle['file.setThumbnail.confirmation']}</span>
+                        </p>
+                    </div>
+                    <div class="button-block">
+                        <p:commandButton id="fileThumbnailConfirmCntBtn"
+                                         value="#{bundle.continue}"
+                                         oncomplete="PF('fileThumbnailConfirm').hide()"
+                                         update=":fileForm:fileTable,:messagePanel"
```

Comment:
I think we can remove :messagePanel from all updates here, it was used to update template error panel but now we have autoUpdate.

Reply:
That is true for global messages (those with id passed equal to null), and not for those created thru `JsfHelper.addFlashMessage()` method like we do in backing bean.

Reply:
But jsf update is used for async based update on the same page, and flash is only for keeping information after redirect/ whole page refresh so I don't think it will work anyway.